server/routes/pt.js

```diff
@@ -114,4 +116,184 @@ router.post('/search', verifyToken, async (req, res) => {
   }
 });
 
+router.post('/addptrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+ 
+    Pt.findByIdAndUpdate(pt_id,
+      {$push: {requests: client_id}},
+      {safe: true, upsert: true},
+      function(err,doc) {
+        if(err){
+          console.log(err);
+        }else{
+          return res.status(OK).send(client_id);
+        }
+      });
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/addclientrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+    
+    Client.findByIdAndUpdate(client_id,
+      {$push: {requests: pt_id}},
+      {safe: true, upsert: true},
+      function(err,doc) {
+        if(err){
+          console.log(err);
+        }else{
+          return res.status(OK).send(pt_id);
+        }
+      });
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/getrequests', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+    const doc = await Pt.findById(id);
+    if (!doc) return res.status(NOT_FOUND).send();
+
+    const requests = doc.requests;
+    const names = [];
+    for await (const docs of User.findById(requests)) {
+      try {
+        const user = await User.findById(docs._id);
+        names.push(user.publicName);
+      }
+      catch (err) {
+        console.error('/getrequests found missing User doc: ' +
+        err.message)
+      }
+    }
+
+    return res.status(OK).send(names);
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/acceptrequest', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+
+    const doc = await Pt.findById(id);
+    const requests = doc.requests;
+    
+    for await (const doc of User.findById(requests)) {
+      try {
+        if(doc.publicName == req.body.name) {
+        
+          // Update the PT's clients array to add the requesting client
```

Comment:
See above comment about using `find()` so that we can change everything all at once and check for error before making changes.